src/main/java/org/opentripplanner/graph_builder/module/osm/WayPropertySet.java

```diff
@@ -176,9 +296,28 @@ public I18NString getCreativeNameForWay(OSMWithTags way) {
     }
     
     /**
-     * Calculate the automobile speed, in meters per second, for this way.
+     * Get the automobile speed, in meters per second, for this way. Try to get value from the cache, otherwise
+     * calculate and store in cache.
      */
     public float getCarSpeedForWay(OSMWithTags way, boolean back) {
+        String wayPropertiesKey = getKeyFromApplicableTags(way, possibleCarSpeedTagValues);
+        // also append back value
+        wayPropertiesKey = String.format("%s;%s=%b", wayPropertiesKey, "back=", back);
+        // check if lookup key found
+        Float cachedSpeed = carSpeedLookup.get(wayPropertiesKey);
+        if (cachedSpeed != null) {
```

Comment:
Potential for using `HashMap.computeIfAbsent(key, mapper)` https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#computeIfAbsent-K-java.util.function.Function-